formulaire/src/main/resources/public/ts/directives/question/question-type/question-type-cursor.ts

```diff
@@ -1,15 +1,51 @@
-import {Directive, ng} from "entcore";
+import {ng} from "entcore";
 import {Question} from "@common/models";
+import {IScope} from "angular";
 
-interface IViewModel {
+interface IQuestionTypeFreetextProps {
     question: Question;
     hasFormResponses: boolean;
-    $onInit(): Promise<void>;
+}
+
+interface IViewModel extends ng.IController, IQuestionTypeFreetextProps {
+    initQuestionTypeCursor(): Promise<void>;
     onChangeStep(newStep: number): void;
 }
 
-export const questionTypeCursor: Directive = ng.directive('questionTypeCursor', () => {
+interface IQuestionTypeFreetextScope extends IScope, IQuestionTypeFreetextProps {
+    vm: IViewModel;
+}
+
+class Controller implements IViewModel {
+    question: Question;
+    hasFormResponses: boolean;
+
+    constructor(private $scope: IQuestionTypeFreetextScope, private $sce: ng.ISCEService) {}
+
+    $onInit = async () : Promise<void> => {
+        await this.initQuestionTypeCursor();
+    }
+
+    $onDestroy = async () : Promise<void> => {}
+
+    initQuestionTypeCursor = async () : Promise<void> => {
```

Comment:
```suggestion
    initQuestionTypeCursor = () : void => {
```
Tu as besoin d'une promesse ?